### PR TITLE
🐽 Fix memory leak in chunked streams handling code (#3631) (BACKPORT)

### PR DESCRIPTION
**Pull request by socketpair**

### CODE REVIEW

**Comment by socketpair** [comment]
@asvetlov @kxepal @webknjaz  Automatic checking has failed not by my changes. Please review/merge.

**Comment by kxepal** [comment]
cc @webknjaz 
Seems like some build issues (cython missing). Do you have right solution for this?

**Comment by webknjaz** [comment]
@kxepal it's still in progress. I've postponed its resolution for a while since I didn't have time for it but it's still on my radar. cc #3581.

File: tests/test_streams.py [hunk 1]
```diff
@@ -30,6 +35,38 @@ def protocol():
...
+    mock.Mock,
+    abc.ABCMeta,
+)
+
+
+def get_memory_usage(obj):
+    objs = [obj]
+    # Memory leak may be caused by leaked links to same objects.
+    # Without link counting, [1,2,3] is indistiguishable from [1,2,3,3,3,3,3,3]
+    known = defaultdict(int)
```
Comment (webknjaz) [inline tests/test_streams.py:49]: @socketpair FYI there's `collections.Counter` existing exactly for this use-case :)

**Comment by webknjaz** [comment]
@kxepal JFYI you can go to Travis CI build page and restart individual test jobs manually even if the linting stage is failed.

**Pull request merged**